src/Engine/Graphics/Outdoor.cpp

```diff
@@ -2433,54 +2444,61 @@ void ODM_ProcessPartyActions() {
     }
 
     // walking / running sounds ------------------------
-    if (engine->config->settings.WalkSound.value() && pParty->walk_sound_timer) {
-        if (pParty->walk_sound_timer >= pEventTimer->uTimeElapsed)
-            pParty->walk_sound_timer -= pEventTimer->uTimeElapsed;
-        else
-            pParty->walk_sound_timer = 0;
-    }
-
-    // save up distance deltas so walks sounds play at high fps with small delta
-    int pX_ = pParty->vPosition.x - partyNewX;
-    int pY_ = pParty->vPosition.y - partyNewY;
-    int pZ_ = pParty->vPosition.z - partyNewZ;
-    pParty->_movementTally += integer_sqrt(pX_ * pX_ + pY_ * pY_ + pZ_ * pZ_);
-
-    if (engine->config->settings.WalkSound.value() && pParty->walk_sound_timer <= 0) {
-        pAudioPlayer->StopAll(804);  // stop sound
-        if (partyIsRunning && (!partyNotTouchingFloor || partyCloseToGround)) {
-            if (pParty->_movementTally >= 16) {
-                pParty->_movementTally = 0;
-                if (!partyNotOnModel &&
-                    pOutdoor->pBModels[pParty->floor_face_pid >> 9]
-                    .pFaces[(pParty->floor_face_pid >> 3) & 0x3F].Visible()) {
-                    pAudioPlayer->playWalkSound(SOUND_RunWood);  // бег на 3D Modelи
-                } else {
-                    int v87 = pOutdoor->GetSoundIdByPosition(WorldPosToGridCellX(pParty->vPosition.x), WorldPosToGridCellY(pParty->vPosition.y), 1);
-                    pAudioPlayer->playWalkSound((SoundID)v87);  // бег по земле 56
+    if (engine->config->settings.WalkSound.value()) {
+        pParty->walk_sound_timer -= pEventTimer->uTimeElapsed;
+
+        if (pParty->walk_sound_timer <= 0) {
+            pAudioPlayer->stopWalkingSounds();
+        }
+
+        // Start sound processing only when actual movement is performed to avoid
+        // stopping sounds on high FPS
+        if (pEventTimer->uTimeElapsed) {
+            int walkDelta = integer_sqrt((pParty->vPosition - Vec3i(partyNewX, partyNewY, partyNewZ)).lengthSqr());
```

Comment:
Have you tested this over a range of FPS limits?

Reply:
I did set 30FPS and run around, all was okay.
I also think that these limits may not be quite needed because walk/run modes are resolved earlier in that function and only thing I can think of where it is applied is moving on non-horizontal surfaces.